packages/mobile/src/harmony-native/components/Button/Button/Button.tsx

```diff
@@ -0,0 +1,266 @@
+    backgroundColor:
+      type === 'dark' ? 'rgba(50, 51, 77, 0.6)' : 'rgba(255, 255, 255, 0.85)',
+    borderWidth: 1,
+    borderStyle: 'solid',
+    borderColor: themeColors.border.default,
+
+    // TODO: Need to add blur
+    // Could use BlurView or try something native
+    // backdropFilter: 'blur(6px)',
+
+    ...(isDisabled && {
+      opacity: 0.45
+    })
+  }
+  const tertiaryDynamicStyles = {
+    default: {
+      background: type === 'dark' ? '#32334d99' : '#ffffffd9',
+      text: themeColors.text.default,
+      border: themeColors.border.default
+    },
+    press: {
+      background: themeColors.background.surface2,
+      text: themeColors.text.default,
+      border: themeColors.border.strong
+    }
+  }
+
+  const destructiveStyles: ReactNativeStyle = {
+    backgroundColor: 'transparent',
+    borderWidth: 1,
+    borderStyle: 'solid',
+    borderColor: themeColors.special.red,
+
+    ...(isDisabled && {
+      opacity: 0.45
+    })
+  }
+  const destructiveDynamicStyles = {
+    default: {
+      background: 'transparent',
+      text: themeColors.text.danger
+    },
+    press: {
+      background: themeColors.special.red,
+      text: themeColors.text.staticWhite
+    }
+  }
+
+  const buttonCss: ReactNativeStyle = {
+    backgroundColor: primaryOverrideColor ?? themeColors.primary.primary,
+    borderWidth: 0,
+    borderRadius: cornerRadius.s,
+    alignItems: 'center',
+    justifyContent: 'center',
+    ...shadows.near,
+
+    ...(variant === ButtonType.SECONDARY
+      ? secondaryStyles
+      : variant === ButtonType.TERTIARY
+      ? tertiaryStyles
+      : variant === ButtonType.DESTRUCTIVE
+      ? destructiveStyles
+      : primaryStyles),
+
+    ...(size === ButtonSize.SMALL
+      ? smallStyles
+      : size === ButtonSize.LARGE
+      ? largeStyles
+      : defaultStyles),
+
+    ...(isDisabled && { shadowColor: 'transparent' })
+  }
+
+  const textCss: TextStyle = {
+    ...(size === ButtonSize.SMALL
+      ? smallTextStyles
+      : size === ButtonSize.LARGE
+      ? largeTextStyles
+      : defaultTextStyles)
+  }
+
+  const textColor =
+    (variant === ButtonType.SECONDARY && !isDisabled) ||
+    variant === ButtonType.TERTIARY
+      ? 'default'
+      : variant === ButtonType.DESTRUCTIVE
+      ? 'danger'
+      : 'staticWhite'
+
+  const iconSize: IconProps['size'] =
+    size === ButtonSize.SMALL ? 's' : size === ButtonSize.LARGE ? 'l' : 'm'
+
+  const loaderSize =
+    size === ButtonSize.SMALL ? 16 : size === ButtonSize.LARGE ? 24 : 20
+
+  const dynamicStyles =
+    variant === ButtonType.SECONDARY
+      ? secondaryDynamicStyles
+      : variant === ButtonType.TERTIARY
+      ? tertiaryDynamicStyles
+      : variant === ButtonType.DESTRUCTIVE
+      ? destructiveDynamicStyles
+      : primaryDynamicStyles
+
+  return (
+    <BaseButton
+      disabled={isDisabled}
+      style={buttonCss}
+      dynamicStyles={isDisabled ? undefined : dynamicStyles}
+      innerProps={{
```

Comment:
Just checking that we have this pattern in web? Or is it the case in web we just have css do the modifications?

Reply:
yea doing this bc its easier to apply styles to inner elements in web. Need to pass along fill colors and things on mobile. Not the most fun, but wanted to find a setup that didn't change the api of the exported components too much